airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -65,6 +97,12 @@ def _choose_wider_type(key: str, t1: JsonSchemaSupportedType, t2: JsonSchemaSupp
         raise SchemaInferenceError(FileBasedSourceError.UNRECOGNIZED_TYPE, key=key, detected_types=f"{t1},{t2}")
 
 
+def is_equal_or_narrower_type(value: Any, expected_types: List[str]):
```

Comment:
That's a good point. I actually did some substantial reworking & testing of the functions in this file in the latest commit, because there was some dummy code in there that wasn't being exercised since we're only supporting CSVs/string types currently. When adding test coverage I found and fixed a couple of bugs.